src/families/cosmos/js-signOperation.ts

```diff
@@ -32,52 +31,39 @@ const signOperation = ({
       let cancelled;
 
       async function main() {
+        const hwApp = new Cosmos(transport);
+
         const { accountNumber, sequence } = await getAccount(
           account.freshAddress
         );
 
         const chainId = await getChainId();
 
-        const hdPaths: any = stringToPath("m/" + account.freshAddressPath);
-
-        const ledgerSigner = new LedgerSigner(transport, {
-          hdPaths: [hdPaths],
-        });
-
         o.next({ type: "device-signature-requested" });
 
-        const accounts = await ledgerSigner.getAccounts();
-
-        let pubkey;
+        const { publicKey } = await hwApp.getAddress(
+          account.freshAddressPath,
+          "cosmos",
+          false
+        );
```

Comment:
Sorry I didn't think of this before, but can't the public key be deduced from `account.freshAddress`?
Same as done [here](https://github.com/LedgerHQ/ledger-live-common/blob/7ff03d79cd5c812a840a15c55ab8e97a3a2c7faf/src/families/cosmos/js-synchronisation.ts#L167).
This would save time without the extra call to the Nano app, if we can deduce the info.

Reply:
I don't think it is available without the full signature package, decodeBech32Pubkey is exclusively for a bech32 format.
By the way, it is the reverse operation that is done [here](https://github.com/LedgerHQ/ledger-live-common/blob/7ff03d79cd5c812a840a15c55ab8e97a3a2c7faf/src/families/cosmos/js-synchronisation.ts#L167) (pubkey to address) which is much simpler.
Not using the hwApp will add CosmJS dependency cost, isn't this contradictory with protobuf switching?